src/umpire/strategy/DynamicPool.hpp

```diff
@@ -39,129 +38,117 @@ namespace strategy {
  * and the minimum size controls the lower bound on all future chunk
  * allocations.
  */
-class DynamicPool :
-  public AllocationStrategy
+class DynamicPool : public AllocationStrategy
 {
   public:
+    using Pointer = void*;
+
     /*!
-     * \brief Callback Heuristic to trigger coalesce of free blocks in pool.
+     * \brief Callback heuristic to trigger coalesce of free blocks in pool.
      *
      * The registered heuristic callback function will be called immediately
      * after a deallocation() has completed from the pool.
      */
-    using Coalesce_Heuristic = std::function<bool( const strategy::DynamicPool& )>;
+    using CoalesceHeuristic = std::function<bool (const strategy::DynamicPool&)>;
 
     /*!
      * \brief Construct a new DynamicPool.
      *
-     * \param name Name of this instance of the DynamicPool.
-     * \param id Id of this instance of the DynamicPool.
-     * \param min_initial_alloc_size The minimum size of the first allocation
-     *                               the pool will make.
-     * \param min_alloc_size The minimum size of all future allocations.
-     * \param coalesce_heuristic Heuristic callback function.
+     * \param name Name of this instance of the DynamicPool
+     * \param id Unique identifier for this instance
+     * \param initial_alloc_bytes Size the pool initially allocates
+     * \param min_alloc_bytes The minimum size of all future allocations
+     * \param align_bytes Number of bytes with which to align allocation sizes
+     * \param coalesce_heuristic Heuristic callback function
      */
     DynamicPool(
         const std::string& name,
         int id,
         Allocator allocator,
-        const std::size_t min_initial_alloc_size = (512 * 1024 * 1024),
+        const std::size_t initial_alloc_size = (512 * 1024 * 1024),
         const std::size_t min_alloc_size = (1 * 1024 *1024),
-        Coalesce_Heuristic coalesce_heuristic = heuristic_percent_releasable(100)) noexcept;
+        const int align_bytes = 16,
```

Comment:
Adding this new default argument in this position breaks current users. I think we should have multiple constructors that delegate down to this one. This way we can have one with alignment and default heuristic, and one with default alignment and a heuristic.

Reply:
Yes, I was worried about backward compatibility. Good idea 👍